### DISCUSSION TITLE
Any plans for a Security Framework like the ones provided for Asp.Net Core / Blazor developers?

### DISCUSSION THREAD

**Discussion by githubfanster** [dotnet/maui#208] [General]
I'm talking something similar to the AuthorizeAttribute, AuthorizeView, Role- and Policy-Based Authorization, and the ability to check authorization rules as part of procedural logic (e.g, @inject IAuthorizationService AuthorizationService ... if ((await AuthorizationService.AuthorizeAsync(user, "content-editor")).Succeeded) { ... } described in https://docs.microsoft.com/en-us/aspnet/core/blazor/security/?view=aspnetcore-3.1.

I come from a WPF background, and when I started learning Asp.Net Core and Blazor, I was so pleased to find out that the built-in security support facilitates development productivity. I'm new to Xamarin.Form, so this question might be showing my ignorance, but the research I've initiated has so far left me with the impression that aside from an ability to define CustomPermission(s) similar to the ones made available in Xamarin.Essentials, out of the box Xamarin.Forms does not provide other conveniences for coding business-rule-related authorization logic.

I'm not sure if defining CustomPermission(s) similar to the ones in Xamarin.Essentials would do the job for me (can they be used to adequately define business-rule-related authorization logic, or are they more restricted to defining permission logic for using device features only), so could some kind soul help enlighten me and let me know if my question adds value to the MAUI "wants" discussion? If there's been an oversight to my research, could you also please point me to some urls?

Thank you so much!

**Comment by listepo** [comment]
I agree it would be nice to have.

**Comment by dan5602** [comment]
I am looking for some advice options on how people are currently managing roles/permissions in MAUI. 
I have MAUI iOS/Android companion apps that communicate with a dotnet web API that returns a JWT token on login which contains roles/permissions. 
What is the recommended way of showing/hiding UI based on roles/permissions contained in the JWT token?
I would be interested to see how others are doing this.
